spec/lib/data_update_scripts/append_collective_noun_to_community_name_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+require Rails.root.join(
+  "lib/data_update_scripts/20201228194641_append_collective_noun_to_community_name.rb",
+)
+
+describe DataUpdateScripts::AppendCollectiveNounToCommunityName do
+  context "when collective_noun_disabled is false" do
+    it "appends the collective_noun to the community_name" do
+      SiteConfig.collective_noun = "Club"
```

Comment:
Rather than explicitly assigning a value to a field on `SiteConfig` here, I think we should opt to use Rspec's [method stubs](https://relishapp.com/rspec/rspec-mocks/v/2-99/docs/method-stubs) in this test and the one below, too.

By using a stub here, we can be sure that we're not accidentally ever changing the value of `SiteConfig.collective_noun`, which could leak into other tests or cause strange, confusing bugs in tests down the road.

In this test, that would look something like:
```ruby
allow(SiteConfig).to receive(:collective_noun).and_return("Club")
```

Reply:
Ahh, that's great! I could also just lean into the default `collective_noun`, which is "Community"--I set `collective_noun` to "Club" here just to make the test more explicit for folks who look at this code in the future. Thoughts @vaidehijoshi ?

Reply:
That would work!

But, there's one thing to keep in mind here, which can a bit obtuse until you think about how this test might be interpreted in the future. There is one _benefit_ of being explicit in the test, like the way you currently have it. The benefit is that if the (default) value you are testing against changes down the road (aka someone goes and changes [this line](https://github.com/forem/forem/pull/12055/files#diff-68da686a0f7348bf9a2c59e669043f21019ab8dfb6363cea6daa534c5c8296d7R58)), then this test would start failing. Sometimes, you _want_ your test to fail in that way, and relying on the default makes sense.

But this test isn't really testing the default value of `community_club`; it's testing the behavior of the _data update script_. So, if you were _not_ explicit about the value of this field in this test and if this fails down the road, it could be confusing (at first) to the person making the change and they'd have to figure out that the test was failing because it was _implicitly_ depending upon + testing the default value of the SiteConfig's field. By being explicit here, you are actually avoiding that scenario!

TL;DR, I would vote to be explicit, just as you have it set up already 😉

Reply:
Thank you so much for the thorough explanation! 😄 

I just pushed up changes that incorporated method stubs in the spec before seeing this comment @vaidehijoshi . 😂 I think the stubs sound like a safer way forward, even if they're not as explicit. That being said, would it be helpful if I added a comment to the spec now that I've changed it to give future contributors/us a bit more context?

Reply:
> I think the stubs sound like a safer way forward, even if they're not as explicit.

@juliannatetreault I think maybe there is a little confusion here: the stubs don't actually have anything to do with the explicitness that I was talking about in my comment. What makes this test explicit is when you declare that `collective_noun` is set to return a value of the string `"Club"`:
```ruby
# explicitly setting the collective_noun by using a method stub
allow(SiteConfig).to receive(:collective_noun).and_return("Club")
```

To be clear, what you had in your initial commit was _also explicit_, even though it didn't use a stub:
```ruby
# explicitly setting the collective_noun by setting the value directly, which is less safe
SiteConfig.collective_noun = "Club"
```
Both of these are explicit because you are explicitly setting the value of `collective_noun` (either with a method stub, or directly) in the body of the test. When someone reads either of those two variations above,  

The _implicitness_ I was talking about would have only applied if you had relied on the default value, like you were describing in your earlier comment:
> Alternatively, I could also just lean into the default value for collective_noun, which is "Community"

So, if you had written your test like this:
```ruby
described_class.new.run
expect(SiteConfig.collective_noun_disabled).to eq(false)
expect(SiteConfig.community_name).to eq("DEV(local) Community")
```

then _that_ would have been implicit, because you would have been implicitly relying on the fact that the default value of `collective_noun` is `"Community"`. 

Hopefully that explanation makes more sense? :) Let me know if any of it is still confusing. Given all of that, I don't think you need to add a comment here because the test is being explicit while using a stub 👍🏽

Reply:
OH! I misread your comment, @vaidehijoshi -- apologies. 😊 I'll leave everything as-is then.